provider/datagen/src/options.rs

```diff
@@ -6,19 +6,30 @@
 
 pub use crate::transform::cldr::source::CoverageLevel;
 
-/// TODO
+/// Defines how fallback will apply to the generated data.
 #[derive(Debug, PartialEq, Clone, Copy, serde::Serialize, serde::Deserialize)]
 #[non_exhaustive]
 pub enum FallbackMode {
-    /// TODO
-    None,
-    /// TODO
-    Full,
+    /// This mode tries to generate data for the supplied locales. If data doesn't exist for a locale, it will be skipped.
+    ///
+    /// This is the pre-1.2 behavior, and requires manual runtime fallback.
+    Legacy,
```

Comment:
I don't think we've been consistent enough in `supported_locales` to call this passthrough. It's a weird mode so I think `Legacy` is a more appropriate name.

Other than name bikeshedding, what's your thought on these three modes?

Reply:
> I don't think we've been consistent enough in supported_locales

I think `supported_locales` is implemented accurately; is it not?

`Expand` mode seems fine. `Runtime` is okay but I think there are two ways that it could run:

1. All Locales: same as `PassThrough` but prune entries reachable from fallback
2. Filtered Locales: same as `Expand` but prune entries reachable from fallback (as well as entries not reachable from any explicitly listed locale)

Reply:
(2) sounds like the more expensive version of (1). If you're doing runtime fallback anyway, there won't be a difference?

Reply:
Let's discuss this later after this PR is landed